stackrox/stackrox
# Delete classic Search modal because superseded by Search page

## Description

As a prerequisite to simplified solution to banner layout problems, remove superseded Search modal that depended on absolute position.

### Files changed

1. Remove `ROX_SEARCH_PAGE_UI` feature flag:
    * pkg/features/list.go
    * tests/e2e/lib.sh
    * ui/apps/platform/src/types/featureFlag.ts

2. Remove conditional rendering:
    * src/Containers/MainPage/Body.tsx
    * src/Containers/MainPage/Header/GlobalSearchButton.tsx
    * src/Containers/MainPage/MainPage.tsx

3. Update style:
    * src/css/acs.css add 1 rule
    * src/app.tw.css delete 2 rules

4. Update reducers and sagas
    * src/reducers/index.js
    * src/sagas/index.js

### Files deleted

1. Integration test
    * cypress/integration/search.test.js

2. Classic files in src/Containers/Search folder
    * ReduxSearchInput.js
    * SearchAutoComplete folder
    * SearchModal.js
    * SearchResults.tsx
    * searchCategoryDescriptorMap.ts

3. Reducers
    * src/reducers/globalSearch.js

4. Sagas
    * src/sagas/globalSearchSagas.js
    * src/sagas/searchSagas.js
    * src/sagas/searchSagas.test.js

### Residue

1. Only remaining occurrence of src/reducers/pageSearch.ts is in classic Network Graph
2. Investigate whether the same applies to ui/apps/platform/src/sagas/searchAutocompleteSagas.js

## Checklist
- [x] Deleted obsolete integration tests

## Testing Performed

1. Click **Search** on masthead to open Search page without feature flag.
2. See link color override rule in Elements pane and see underline on hover, but decide no reason to override that browser default.

![Search_link_inherit_and_underline](https://user-images.githubusercontent.com/11862657/218791984-2f47d85b-219f-4cfb-b745-ca8353f38304.png)
